### PR TITLE
run rustfmt on libsyntax_ext folder

### PR DESCRIPTION
**Pull request by srinivasreddy**

### CODE REVIEW

**Comment by srinivasreddy** [comment]
cc @nrc  take a look

File: src/libsyntax_ext/asm.rs [hunk 1]
```diff
@@ -30,43 +29,48 @@ enum State {
...
     // Split the tts before the first colon, to avoid `asm!("x": y)`  being
     // parsed as `asm!(z)` with `z = "x": y` which is type ascription.
-    let first_colon = tts.iter().position(|tt| {
-        match *tt {
-            ast::TokenTree::Token(_, token::Colon) |
-            ast::TokenTree::Token(_, token::ModSep) => true,
-            _ => false
-        }
-    }).unwrap_or(tts.len());
+    let first_colon = tts.iter()
```
Comment (aturon) [inline src/libsyntax_ext/asm.rs]: @nrc I would love if rustfmt would avoid this kind of indentation (when you have a multi-line closure in the middle of a method chain).

File: src/libsyntax_ext/env.rs [hunk 2]
```diff
@@ -8,11 +8,10 @@
...
 
-/*
- * The compiler code necessary to support the env! extension.  Eventually this
- * should all get sucked into either the compiler syntax extension plugin
- * interface.
- */
+// The compiler code necessary to support the env! extension.  Eventually this
+// should all get sucked into either the compiler syntax extension plugin
+// interface.
+//
```
Comment (aturon) [inline src/libsyntax_ext/env.rs]: This extra `//` looks like a bug to me, @nrc.

File: src/libsyntax_ext/format.rs [hunk 3]
```diff
@@ -410,17 +409,38 @@ impl<'a, 'b> Context<'a, 'b> {
                 let prec = self.trans_count(arg.format.precision);
                 let width = self.trans_count(arg.format.width);
                 let path = self.ecx.path_global(sp, Context::rtpath(self.ecx, "FormatSpec"));
-                let fmt = self.ecx.expr_struct(sp, path, vec!(
-                    self.ecx.field_imm(sp, self.ecx.ident_of("fill"), fill),
-                    self.ecx.field_imm(sp, self.ecx.ident_of("align"), align),
-                    self.ecx.field_imm(sp, self.ecx.ident_of("flags"), flags),
-                    self.ecx.field_imm(sp, self.ecx.ident_of("precision"), prec),
-                    self.ecx.field_imm(sp, self.ecx.ident_of("width"), width)));
+                let fmt = self.ecx.expr_struct(sp,
```
Comment (aturon) [inline src/libsyntax_ext/format.rs]: Ugh! This is another example where rustfmt tends toward serious rightward drift.

**Comment by aturon** [comment]
I think this is basically fine to land, but has a couple examples where rustfmt ends up pushing things *far* to the right. It'd be great to have some rules that can avoid this (though I imagine it's not easy).

File: src/libsyntax_ext/asm.rs [hunk 4]
```diff
@@ -30,43 +29,48 @@ enum State {
...
-            Clobbers  => Options,
-            Options   => StateNone,
-            StateNone => StateNone
+            Asm => Outputs,
+            Outputs => Inputs,
+            Inputs => Clobbers,
+            Clobbers => Options,
+            Options => StateNone,
+            StateNone => StateNone,
         }
```
Comment (tshepang) [inline src/libsyntax_ext/asm.rs]: is removing the alignment like this ok?

File: src/libsyntax_ext/asm.rs [hunk 5]
```diff
@@ -108,9 +113,7 @@ pub fn expand_asm<'cx>(cx: &'cx mut ExtCtxt, sp: Span, tts: &[ast::TokenTree])
                 asm_str_style = Some(style);
             }
             Outputs => {
-                while p.token != token::Eof &&
-                      p.token != token::Colon &&
-                      p.token != token::ModSep {
+                while p.token != token::Eof && p.token != token::Colon && p.token != token::ModSep {
```
Comment (tshepang) [inline src/libsyntax_ext/asm.rs]: this is less readable

File: src/libsyntax_ext/asm.rs [hunk 6]
```diff
@@ -155,9 +157,7 @@ pub fn expand_asm<'cx>(cx: &'cx mut ExtCtxt, sp: Span, tts: &[ast::TokenTree])
                 }
             }
             Inputs => {
-                while p.token != token::Eof &&
-                      p.token != token::Colon &&
-                      p.token != token::ModSep {
+                while p.token != token::Eof && p.token != token::Colon && p.token != token::ModSep {
```
Comment (tshepang) [inline src/libsyntax_ext/asm.rs]: less readable

File: src/libsyntax_ext/asm.rs [hunk 7]
```diff
@@ -179,9 +179,7 @@ pub fn expand_asm<'cx>(cx: &'cx mut ExtCtxt, sp: Span, tts: &[ast::TokenTree])
                 }
             }
             Clobbers => {
-                while p.token != token::Eof &&
-                      p.token != token::Colon &&
-                      p.token != token::ModSep {
+                while p.token != token::Eof && p.token != token::Colon && p.token != token::ModSep {
```
Comment (tshepang) [inline src/libsyntax_ext/asm.rs]: also less readable

File: src/libsyntax_ext/format.rs [hunk 8]
```diff
@@ -460,15 +481,11 @@ impl<'a, 'b> Context<'a, 'b> {
...
-                ast::Mutability::Immutable);
-        let pieces = Context::static_array(self.ecx,
-                                           "__STATIC_FMTSTR",
-                                           piece_ty,
-                                           self.str_pieces);
+        let piece_ty = self.ecx.ty_rptr(self.fmtsp,
+                                        self.ecx.ty_ident(self.fmtsp, self.ecx.ident_of("str")),
+                                        Some(static_lifetime),
+                                        ast::Mutability::Immutable);
+        let pieces = Context::static_array(self.ecx, "__STATIC_FMTSTR", piece_ty, self.str_pieces);
```
Comment (tshepang) [inline src/libsyntax_ext/format.rs]: I much prefer the unchanged formatting here... if one decides to change, for example, name of `piece_ty`, they won't also need to re-align the other lines.

**Comment by alexcrichton** [comment]
Closing due to inactivity, but feel free to resubmit with a rebase!

**Pull request closed without merging**